app/src/main/java/io/github/louistsaitszho/lineage/activities/MainActivity.kt

```diff
@@ -2,12 +2,113 @@ package io.github.louistsaitszho.lineage.activities
 
 import android.support.v7.app.AppCompatActivity
 import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.Menu
+import android.view.MenuItem
+import io.github.louistsaitszho.lineage.ItemListUnit
 import io.github.louistsaitszho.lineage.R
+import io.github.louistsaitszho.lineage.RecyclerViewAdapter
 
 class MainActivity : AppCompatActivity() {
 
+
+    //This is the way to declare Static final in Kotlin
+    companion object{
+        const val MEDIUM = "Medium"
+        const val LARGE = "Large"
+        const val NO_THUMBNAIL = "No Thumbnail"
+
+    }
+
+
+
+    private var layoutOption: String? = null
+
+    //RecyclerView
+    private var recyclerView: RecyclerView? = null
```

Comment:
take a look at https://kotlinlang.org/docs/tutorials/android-plugin.html. in kotlin this allows you to skip those ugly findviewbyid